tests/remote-schema/remote-schema.test.ts

```diff
@@ -0,0 +1,510 @@
+            password: string;
+          };
+        };
+        responses: {
+          /** successful operation */
+          200: {
+            headers: {
+              /** calls per hour allowed by the user */
+              "X-Rate-Limit"?: number;
+              /** date in UTC when token expires */
+              "X-Expires-After"?: string;
+            };
+            content: {
+              "application/xml": string;
+              "application/json": string;
+            };
+          };
+          /** Invalid username/password supplied */
+          400: unknown;
+        };
+      };
+      logoutUser: {
+        responses: {
+          /** successful operation */
+          default: unknown;
+        };
+      };
+      getUserByName: {
+        parameters: {
+          path: {
+            /** The name that needs to be fetched. Use user1 for testing. */
+            username: string;
+          };
+        };
+        responses: {
+          /** successful operation */
+          200: {
+            content: {
+              "application/xml": external["https://raw.githubusercontent.com/drwpow/openapi-typescript/main/tests/v3/specs/petstore.yaml"]["components"]["schemas"]["User"];
+              "application/json": external["https://raw.githubusercontent.com/drwpow/openapi-typescript/main/tests/v3/specs/petstore.yaml"]["components"]["schemas"]["User"];
+            };
+          };
+          /** Invalid username supplied */
+          400: unknown;
+          /** User not found */
+          404: unknown;
+        };
+      };
+      /** This can only be done by the logged in user. */
+      updateUser: {
+        parameters: {
+          path: {
+            /** name that need to be updated */
+            username: string;
+          };
+        };
+        responses: {
+          /** Invalid user supplied */
+          400: unknown;
+          /** User not found */
+          404: unknown;
+        };
+        /** Updated user object */
+        requestBody: {
+          content: {
+            "*/*": external["https://raw.githubusercontent.com/drwpow/openapi-typescript/main/tests/v3/specs/petstore.yaml"]["components"]["schemas"]["User"];
+          };
+        };
+      };
+      /** This can only be done by the logged in user. */
+      deleteUser: {
+        parameters: {
+          path: {
+            /** The name that needs to be deleted */
+            username: string;
+          };
+        };
+        responses: {
+          /** Invalid username supplied */
+          400: unknown;
+          /** User not found */
+          404: unknown;
+        };
+      };
+    };
+  };
+  "subschema/remote1.yml": {
+    paths: {};
+    components: {
+      schemas: {
+        /** this is a duplicate of spec.yml#components/schemas/Remote1 */
+        Remote1: string;
+        Remote2: external["subschema/remote2.yml"]["components"]["schemas"]["Remote2"];
+        Circular: components["schemas"]["Circular"];
```

Comment:
If you have a circular reference back to the original schema, that’s preserved! 🎉